src/prpy/planning/base.py

```diff
@@ -466,6 +466,96 @@ def plan(self, method, args, kw_args):
         else:
             raise UnsupportedPlanningError()
 
+
+class HardwareJointLimitValidator(MetaPlanner):
+    def __init__(self, planner):
+        super(HardwareJointLimitValidator, self).__init__()
+        self._planner = planner
+
+    def __str__(self):
+        return 'HardLimitValidator({:s})'.format(', '.join(map(str, self._planners)))
+
+    def get_planners(self, method_name):
+        if planner.has_planning_method(method_name):
+            return [self._planner]
+        else:
+            return []
+
+    def plan(self, method, args, kw_args):
+
+        # TODO : Is this correct?
+        robot = args[0]
+
+        #Get HARD DOF Limits from robot
+        #hard_limit_min, hard_limit_max = robot.GetHardwareDOFLimits()
+        active_dof_indices = robot.GetActiveDOFIndices()
+        currentDOFVals = robot.getActiveDOFValues()
+        soft_limit_min, soft_limit_max = robot.GetActiveDOFLimits()
+        resetLimits = False
+
+        lower_hard_violations = (currentDOFVals < hard_limit_min)
+        upper_hard_violations = (currentDOFVals > hard_limit_max)
+        lower_soft_violations = (currentDOFVals < soft_limit_min)
+        upper_soft_violations = (currentDOFVals > soft_limit_max)
```

Comment:
I'm assuming that `robot.GetHardwareDOFLimits` will return `(-inf, inf)` if there is no limit. Otherwise, this will need to handle some special cases.